src/app/util/af-event.cpp

```diff
@@ -73,28 +65,39 @@ void emberAfPluginIasZoneClientStateMachineEventHandler(void){};
 EMBER_AF_GENERATED_EVENT_CODE
 #endif // EMBER_AF_GENERATED_EVENT_CODE
 
-#if defined(EMBER_AF_GENERATED_EVENT_CONTEXT)
-uint16_t emAfAppEventContextLength        = EMBER_AF_EVENT_CONTEXT_LENGTH;
-EmberAfEventContext emAfAppEventContext[] = { EMBER_AF_GENERATED_EVENT_CONTEXT };
-#endif // EMBER_AF_GENERATED_EVENT_CONTEXT
-
-const char * emAfEventStrings[] = {
-
-#ifdef EMBER_AF_GENERATED_EVENTS
-    EMBER_AF_GENERATED_EVENT_STRINGS
-#endif
-
-    nullptr,
+struct MatterEventMetaContext
+{
+    EmberAfEventContext context;
+    const char * eventString;
+    EmberEventData event;
+    MatterEventMetaContext * nextContext;
 };
 
-EmberEventData emAfEvents[] = {
+MatterEventMetaContext * metaContextListHead = nullptr;
 
-#ifdef EMBER_AF_GENERATED_EVENTS
-    EMBER_AF_GENERATED_EVENTS
-#endif
+void MatterEventMetaContextAppend(MatterEventMetaContext & newContext)
```

Comment:
There is no mechanism to free any of these, so it's a mandatory leak. This will cause failures of validation of valgrind on an all-clusters-app when we clean-up the shutdown.

Please add a mechanism to clean it up